### DISCUSSION TITLE
GSoC Proposal - Interactivity and visualization tools

### DISCUSSION THREAD

**Discussion by luyuzhe111** [JuliaImages/Images.jl#943] [JSoC]
Hi! I am Bryan, a junior majoring in maths and computer science at Vanderbilt University. I have been doing research in medical imaging, data visualizations, and deep learning. This project is listed as open-ended yet I feel like it might be really related to what I am interested in. So, I am just wondering whether there is any specific algorithm or interface that I can start with. Thanks!

**Comment by johnnychen94** [comment]
I guess you're talking about making [ImageViews.jl](https://github.com/JuliaImages/ImageView.jl) a better visualization tool by adding more convenient helpers/buttons. `ImageViews.jl` is built on top of [Gtk.jl](https://github.com/JuliaGraphics/Gtk.jl), which is a Julia wrapper of the [GTK](https://www.gtk.org/) GUI library. Doing this would be easier if you have GUI development experiences.

For what it worths, another similar toolbox to GTK is [`Makie.jl`](https://github.com/JuliaPlots/Makie.jl), which is a GPU enabled plot framework.

**Comment by luyuzhe111** [comment]
Hi Johnny, thank you so much for your advice and explanations! I will definitely check out the repositories you mentioned! Also, to be more specific, I am referring to this project [Interactivity and visualization tools](https://julialang.org/jsoc/gsoc/images/#interactivity_and_visualization_tools_open-ended).

**Comment by johnnychen94** [comment]
I think @timholy's original expectation is to bring more love to ImageViews.jl. But still, depending on how ambitious you're, if you want to rebuild the whole ImageViews functionality on top of Makie or other similar GUI framework, go ahead and try it. I don't have GUI experiences so this is all the advice I can give.

BTW, if it's only related to visualization, you can also take a look at ImageShow, ImageInTerminal, and ImageDraw and probably get some detailed project ideas from them.

**Comment by luyuzhe111** [comment]
Thank you so much Johnny! This information is super helpful! I will look into them and see what inspirations I can get 😄.
